homework/hackyourtemperature/__tests__/app.test.js

```diff
@@ -0,0 +1,43 @@
+import app from "../app.js";
+import supertest from "supertest";
+
+const request = supertest(app);
+
+describe("When app is open", () => {
+  it("It must responds 200 status code.", async () => {
+    const response = await request
+      .get("/")
+    expect(response.statusCode).toBe(200);
+  })
+})
+
+describe("if requested city name is VALID", () => {
+  it("Valid city name responds 200 status code.", async () => {
+    const response = await request
+      .post("/weather")
+      .send({ cityName: "Almelo" });
+    expect(response.statusCode).toBe(200);
+  })
+})
+
+
+describe("if requested city name is INVALID", () => {
+  it("Invalid city name responds 400 status code.", async () => {
+    const response = await request
+      .post("/weather")
+      .send({ cityName: "ndnd" });
+    expect(response.statusCode).toBe(400);
+  })
+})
+
+describe("if requested city name is INVALID", () => {
+  it("Error message will be appeared", async () => {
+    const response = await request
+      .post("/weather")
+      .send({ cityName: "ndnd" });
+    const errorMessage = response.text.includes("City is not found!");
+    expect(errorMessage).toBe(true);
+  })
+})
```

Comment:
I would combine these since they are testing two things happen in one scenario.

```suggestion
describe("if requested city name is INVALID", () => {
  it("Invalid city name responds 400 status code.", async () => {
    const response = await request
      .post("/weather")
      .send({ cityName: "ndnd" });
    expect(response.statusCode).toBe(400);
    const errorMessage = response.text.includes("City is not found!");
    expect(errorMessage).toBe(true);
  })
})
```